security/utils.py

```diff
@@ -312,31 +313,50 @@ def remove_nul_from_string(value):
     return value.replace('\x00', '')
 
 
+def is_atty_string(s):
+    return bool(re.search('^\\x1b\[\d+m$', s))
+
+
 class LogStringIO(StringIO):
 
+    BOLD = '\x1b[1m'
+    RESET = '\x1b[0m'
+
     def __init__(self, post_write_callback=None, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._last_newline = 0
         self._post_write_callback = post_write_callback
+        self._write_time = True
 
     def _post_write(self):
         if self._post_write_callback:
             self._post_write_callback(self)
 
     def write(self, s):
-        line_first = True
-        for line in s.split('\n'):
-            if not line_first:
-                super().write('\n')
-                self._last_newline = self.tell()
+        lines = s.split('\n')
+
+        for i, line in enumerate(lines):
             cursor_first = True
             for cursor_block in line.split('\r'):
                 if not cursor_first:
                     self.seek(self._last_newline)
                     self.truncate()
+                    self._write_time = True
                 cursor_first = False
+
+                if self._write_time and not is_atty_string(cursor_block):
+                    cursor_block = '{}{}[{}]{} {}'.format(
+                        self.RESET, self.BOLD, now().strftime('%d-%m-%Y %H:%M:%S'), self.RESET, cursor_block
```

Comment:
Maybe use https://pypi.org/project/termcolor/?

Reply:
I didn't want to use new extra library instead of constant it is overkill

Reply:
Especially I must here reset colors which send django. So I still will have to use RESET constant